erigon-lib/trie/proof.go

```diff
@@ -31,13 +31,14 @@ import (
 
 // Prove constructs a merkle proof for key. The result contains all encoded nodes
 // on the path to the value at key. The value itself is also included in the last
-// node and can be retrieved by verifying the proof.
+// node and can be retrieved by verifying the proof. Value is returned as well.
 //
 // If the trie does not contain a value for key, the returned proof contains all
 // nodes of the longest existing prefix of the key (at least the root node), ending
 // with the node that proves the absence of the key.
-func (t *Trie) Prove(key []byte, fromLevel int, storage bool) ([][]byte, error) {
+func (t *Trie) Prove(key []byte, fromLevel int, storage bool) ([][]byte, []byte, error) {
```

Comment:
Not necessary to rely on `Prove` to get the storage value as it can easily be retrieved via stateReader.

Reply:
As we discussed, I's suggest not returning the value bytes since we don't want the function to do more than it should. If you need the value either use a StateReader, or write another function to traverse the trie and get the value. I did this on my branch:

```Go
func (t *Trie) GetAccountNode(key []byte) (value *AccountNode, gotValue bool) {
	hex := keybytesToHex(key)
	return t.getAccount(t.RootNode, hex, 0)
}
```